registry.go

```diff
@@ -0,0 +1,64 @@
+package taskrunner
+
+import (
+	"fmt"
+	"sort"
+)
+
+// DefaultRegistry is where tasks are registered to through the package-level functions Add, Group
+// and Tasks.
+var DefaultRegistry = NewRegistry()
+
+// NewRegistry creates a new task registry.
+func NewRegistry() *Registry {
+	return &Registry{
+		definitions: make(map[string]*Task),
+	}
+}
+
+// Registry is an object that keeps track of task definitions.
+type Registry struct {
+	definitions map[string]*Task
+}
+
+type TaskOption func(*Task)
+
+// Add registers a task, failing if the name has already been taken.
+func (r *Registry) Add(t *Task, opts ...TaskOption) *Task {
+	for _, opt := range opts {
+		opt(t)
+	}
+	if _, ok := r.definitions[t.Name]; ok {
+		panic(fmt.Sprintf("Duplicate task registered: %s", t.Name))
+	}
+	r.definitions[t.Name] = t
+	return t
+}
+
+// Group creates a pseudo-task that groups other tasks underneath it.
+func (r *Registry) Group(name string, tasks ...*Task) {
+	r.Add(&Task{
+		Name:         name,
+		Dependencies: tasks,
+	})
```

Comment:
would also like two other things for groupings:
- allow `Run` to be nil (I don't think it's allowed today)
- prevent them from being depended on

(https://github.com/samsarahq/taskrunner/issues/5)

Reply:
1: done
2: documented